ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -335,10 +335,24 @@ private void checkCapacityAndUpdateWriteStatusDelegate(long totalCapacity, long
       }
       //else: maintain current replicaId status if percentFilled between threshold - delta and threshold
     } else {
-      logger.debug("The WriteStatusDelegate is not instantiated");
+      logger.debug("The ReplicaStatusDelegate is not instantiated");
     }
   }
 
+  /**
+   * Update the stopped replicas list based on the state of this store.
+   * @param isStopped whether the store is stopped ({@code true}) or started.
+   * @return {@code true} if StoppedReplicas list has been updated successfully.
+   */
+  public boolean updateStoppedReplicasList(boolean isStopped) {
```

Comment:
is there a reason to have a special function for this as opposed to doing it on `start()` and `shutdown()`?